src/components/edit/exchange-update/__tests__/useFetchExchange.test.js

```diff
@@ -0,0 +1,195 @@
+import { getMetadataByRequest } from '../useFetchExchange.js'
+
+const MOCK_ENGINE = {
+    query: (_, variables) => ({ query: 'query_name', variables }),
+}
+
+describe('getMetadataByRequest', () => {
+    it('chunks dx items', () => {
```

Comment:
Would it make sense to have a test for a case where eg. both `dx` and `ou` are chunked? This would potentially return some duplicate data, but maybe that doesn't really matter?

Reply:
sure. I've added a test where dx, ou, and pe all get chunked. The metadata all gets added into the same object for lookup and yes there is some minor duplication of retrieving the first ou,pe,dx (necessary because analytics requires them all) or there can be some duplication if different requests use the same analytics items. Either way, the metadata returned should be the same.